### DISCUSSION TITLE
Dockge stack does not show containers that are incorporated into the compose file using "include"

### DISCUSSION THREAD

**Discussion by adamadavidson** [louislam/dockge#294] [Feature Request]
### ⚠️ Please verify that this bug has NOT been reported before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/dockge/security/policy)

### Description

When creating a compose-yaml file for a stack using the ["include" element](https://docs.docker.com/compose/multiple-compose-files/include/), Dockge does not show any containers that are not in the base compose.yaml.

So, if I have one container in the compose.yaml file but 3 further containers added to the stack using the ["include" element](https://docs.docker.com/compose/multiple-compose-files/include/), Dockge only shows the one container.

I appreciate this may be more of a feature request than a bug.

### 👟 Reproduction steps

1. Create `compose.yaml`
2. Add `include` element to pull in additional containers from other compose files.

### 👀 Expected behavior

See 4 containers in Dockge

### 😓 Actual Behavior

Only 1 container is shown in Dockge

### Dockge Version

1.3.3

### 💻 Operating System and Arch

Ubuntu 22.04

### 🌐 Browser

Microsoft Edge Version 120.0.2210.61

### 🐋 Docker Version

Docker 24.0.7

**Comment by codersaur** [comment]
Better support for the `include` element would be great all round. I define all my networks in a common `docker-networks.yaml` file which I include in each project stack, but Dockge wants to add an empty `networks: {}` element instead.

**Reply by Wigsinator** [reply]
I've started doing the same. Dockge really doesn't play nice with it, the networks list ends up full of blank elements. I'd really appreciate it if Dockge had better support for stacks spread across multiple files.

**Reply by ChrisonSimtian** [reply]
just started migrating to Dockge and ran into exactly this.
Would be nice to see this feature implemented, as it helps to make bigger stacks a bit easier to structure

**Comment by ChilledSlim** [comment]
This ability to interpret the include: directive is very necessary.  It needs to parse them and allow editing of each included file.

**Comment by fraterns** [comment]
Agreed. my structure looks like this.

/docker
/docker/docker-compose-servername1.yml
/docker/docker-compose-servername2.yml
...
/docker/compose/
/docker/compose/<servername1>/
/docker/compose/<servername2>/
...
/docker/compose/<servername1>/app1.yml
/docker/compose/<servername1>/app2.yml
/docker/compose/<servername1>/app3.yml
...
Contents of docker-compose-servername#.yml looks like this
include:
  ########################### SERVICES
  ########################### HOSTNAME defined in .env file
  - compose/$HOSTNAME/app1.yml
  - compose/$HOSTNAME/app2.yml
  - compose/$HOSTNAME/app3.yml

What this allows is separate master compose for each server which will have the proper includes for that server. The whole directory structure can be sync to every server
 
using the following

alias dcrun='sudo docker compose --profile all -f $DOCKER_FOLDER/docker-compose-$HOSTNAME.yml'
alias dcup='dcrun up -d --build --remove-orphans'
alias dcdown='dcrun down --remove-orphans'

Means that the same commands no matter what server I am on uses the set $HOSTNAME in the environment file.

Based off that the proper server docker-compose-servername#.yml is called, which then includes only the /docker/compose/servername#/app#.yml

Perhaps there is an easier way to manage multiple servers and be able to move/change easily, but this is a simple home lab, no overall cluster management, etc. This way all my compose files for all servers are all together in one directory and version controlled. A simple pull from git and set the environment variable and a server can be brought up.

I know I am not the only one doing this as this is not my original idea. 

I'd really like to move from Portainer, and am very interested in the editing/management of the compose files in Dockage, but from what I've read so far, Dockage would have zero idea what to do with my compose files, master and includes. Changing my compose files to the Dockage methodology would completely obliterate the way I use them. I'd be fine with having to specify to Dockage the 'master' compose for the server it's on, but it then needs to read and handle the includes.

Sorry if long winded. Was very excited to have an alternative to Portainer, but then saw this issue.
Thank you.
